website/static/js/share/utils.js

```diff
@@ -26,57 +30,124 @@ var formatNumber = function(num) {
 
 var loadingIcon = m('img[src=/static/img/loading.gif]',{style: {margin: 'auto', display: 'block'}});
 
-var loadMore = function(vm) {
-    if (vm.query().length === 0) {
+utils.errorState = function(vm){
+    vm.results = null;
+    vm.statsData = undefined;
+    vm.time = 0;
+    vm.count = 0;
+    vm.resultsLoading(false);
+    m.redraw(true);
+    $osf.growl('Error', 'invalid query');
+};
+
+utils.updateVM = function(vm, data) {
+    if (data === null) {
         return;
     }
-    var page = vm.page++ * 10;
-    vm.resultsLoading(true);
-
-    m.request({
-        method: 'get',
-        background: true,
-        url: '/api/v1/share/?' + $.param({
-            from: page,
-            q: vm.query(),
-            sort: 'dateUpdated',
-        })
-    }).then(function(data) {
-        vm.time = data.time;
-        vm.count = data.count;
-        data.results.forEach(function(result) {
-            result.title = scrubHTML(result.title);
-            result.description = scrubHTML(result.description);
-        });
-
-        vm.results.push.apply(vm.results, data.results);
-
-        vm.resultsLoading(false);
-    }).then(m.redraw).then(function() {
-        callbacks.map(function(cb) {cb();});
+    vm.time = data.time;
+    vm.count = data.count;
+    data.results.forEach(function(result) {
+        result.title = utils.scrubHTML(result.title);
+        result.description = utils.scrubHTML(result.description);
+    });    
+    vm.results.push.apply(vm.results, data.results);   
+    m.redraw();
+    $.map(callbacks, function(cb) {
+        cb();
     });
 };
 
-var search = function(vm) {
-    if (vm.query().length === 0) {
-        return;
+utils.loadMore = function(vm) {    
```

Comment:
Since the ```vm``` is essentially a huge bundle of shared state, it would be really easy to add it there (in fact, I had some of this functionality in the ViewModel previously, but moved it after a code review). I think adding it the controllers is just a lot trickier since there are 5 separate controllers that need to talk to each other. Accessing the controllers through the ```vm``` isn't really any different than just attaching functions to it. I'm fine doing whatever, I just want to make sure I don't have to change it again.